lib/core/include/irods/library_features.h

```diff
@@ -54,4 +54,9 @@
 /// \since 4.3.2
 #define IRODS_HAS_API_ENDPOINT_REPLICA_TRUNCATE                              202403L
 
+/// Defined if the development library supports #rc_genquery2.
+///
+/// \since 4.3.2
+#define IRODS_HAS_API_ENDPOINT_GENQUERY2                                     202403L
```

Comment:
My understanding based on previous conversations I have listened in on, is that we will be expanding the functionality of genquery2 in the future, whenever we do so, we will probably want to increment this value. Consider renaming this define to reflect that it is not just a boolean, but actually has a numerical value that the consumer might be interested in checking against.

I don't feel that strongly about this though, so feel free to ignore if you don't find my reasoning compelling.

Reply:
I get what you're saying. There's room for expanding/changing how we approach the naming of these.

Do you have suggestions for a new naming scheme?

Reply:
That could work. Not sure I agree with the `_VERSION`.

How do we deal with the other macros? Do we introduce aliases for them?

Perhaps everything becomes ... `IRODS_FEATURE_*` or `IRODS_LIBRARY_*`.

I'm leaning towards `IRODS_FEATURE_API_ENDPOINT_GENQUERY2` or `IRODS_LIBRARY_API_ENDPOINT_GENQUERY2`.

Reply:
Maybe the prefix should be `IRODS_LIBRARY_FEATURE_*`.

That's a lot to type, but makes the most sense IMO since the macros in the file are specific to the library. Maybe there's an abbreviation that works well too.

C++20 shipped the following scheme.
- `__cpp_<language_feature_phrase>` _(no common prefix, ignoring `__cpp_`)_
- `__cpp_lib_<phrase>` _(for library features)_
- `__has_cpp_attribute(<attribute_token>)` _(for cpp attributes)_

If we decided to introduce macros that aren't specific to the library, those macros would need their own prefix and header file. We'd then provide a convenience header _(possibly called `features.h`)_ that includes all feature test macro headers.

Alternatively, we just stick with the existing naming since it's not broken/horrible and keep it moving.

Thoughts?

Reply:
I like both `IRODS_FEATURE_*` and `IRODS_LIBRARY_FEATURE_*`

Reply:
Ok. I'm going to use `IRODS_LIBRARY_FEATURE_*` since the other one is more generic and could be useful later on.

Reply:
Leaving as is.

Created #7636 for addressing the naming scheme change.